code/modules/mob/living/carbon/human/species_types/ethereal.dm

```diff
@@ -73,8 +83,95 @@
 	UnregisterSignal(C, COMSIG_ATOM_EMAG_ACT)
 	UnregisterSignal(C, COMSIG_ATOM_EMP_ACT)
 	QDEL_NULL(ethereal_light)
+	if(rooting)
+		rooting.Remove(C)
 	return ..()
 
+/datum/action/innate/root
+	name = "Root"
+	desc = "Root into good soil to gain charge."
+	check_flags = AB_CHECK_CONSCIOUS
+	button_icon_state = "plant-22"
+	icon_icon = 'icons/obj/flora/plants.dmi'
+	background_icon_state = "bg_alien"
+
+/datum/action/innate/root/Activate()
+	var/mob/living/carbon/human/H = owner
```

Comment:
i would try to avoid single letter vars, I go to the_human or the_elzuose if nothing better comes to mind